src/lib.ts

```diff
@@ -0,0 +1,320 @@
+					Number((metadata[index % 2 ? 'height' : 'width'] as number) * (Number(value) / 100)).toFixed(0)
+				);
+			}
+		});
+
+		image.extract({
+			left: cropValues[0],
+			top: cropValues[1],
+			width: cropValues[2],
+			height: cropValues[3],
+		});
+	}
+
+	// get zoom value
+	const zoom = parseFloat(args.zoom || '1') || 1;
+
+	// resize
+	if (args.resize) {
+		// apply smart crop if available
+		if (args.crop_strategy === 'smart' && !args.crop) {
+			const cropResize = getDimArray(args.resize);
+			const rotatedImage = await image.toBuffer();
+			const result = await smartcrop.crop(rotatedImage, {
+				width: cropResize[0] as number,
+				height: cropResize[1] as number,
+			});
+
+			if (result && result.topCrop) {
+				image.extract({
+					left: result.topCrop.x,
+					top: result.topCrop.y,
+					width: result.topCrop.width,
+					height: result.topCrop.height,
+				});
+			}
+		}
+
+		// apply the resize
+		args.resize = getDimArray(args.resize, zoom) as number[];
+		image.resize({
+			width: args.resize[0],
+			height: args.resize[1],
+			withoutEnlargement: true,
+			position: (args.crop_strategy !== 'smart' && args.crop_strategy) || args.gravity || 'centre',
+		});
+	} else if (args.fit) {
+		const fit = getDimArray(args.fit, zoom) as number[];
+		image.resize({
+			width: fit[0],
+			height: fit[1],
+			fit: 'inside',
+			withoutEnlargement: true,
+		});
+	} else if (args.lb) {
+		const lb = getDimArray(args.lb, zoom) as number[];
+		image.resize({
+			width: lb[0],
+			height: lb[1],
+			fit: 'contain',
+			// default to a black background to replicate Photon API behavior
+			// when no background colour specified
+			background: args.background || 'black',
+			withoutEnlargement: true,
+		});
+	} else if (args.w || args.h) {
+		image.resize({
+			width: Number(args.w) * zoom || undefined,
+			height: Number(args.h) * zoom || undefined,
+			fit: args.crop ? 'cover' : 'inside',
+			withoutEnlargement: true,
+		});
+	}
+
+	// set default quality slightly higher than sharp's default
+	if (!args.quality) {
+		args.quality = applyZoomCompression(82, zoom);
+	}
+
+	// allow override of compression quality
+	if (args.webp) {
+		image.webp({
+			quality: Math.round(clamp(args.quality, 0, 100)),
+		});
+	} else if (metadata.format === 'jpeg') {
+		image.jpeg({
+			quality: Math.round(clamp(args.quality, 0, 100)),
+		});
+	}
+
+	// send image
+	return new Promise((resolve, reject) => {
+		image.toBuffer(async (err, data, info) => {
+			if (err) {
+				reject(err);
+			}
+			// Pass PNG images through PNGQuant as Sharp is not good at compressing them.
+			// See https://github.com/lovell/sharp/issues/478
+			if (info.format === 'png' && metadata.pages! > 1) {
+				data = await imageminPngquant()(data);
+				// Make sure we update the size in the info, to reflect the new
+				// size after lossless-compression.
+				info.size = data.length;
+			}
+
+			// add invalid args
+			resolve({ data, info: { ...info, errors: errors.join(';') } });
```

Comment:
Why not return `errors: string[]`?

Reply:
That's just what it did already and I didn't want to change the interface